src/lib/util/get-currently-triggering-noise-thresholds.ts

```diff
@@ -0,0 +1,113 @@
+import { DateTime } from "luxon"
+import { ZonedTime } from "zoned-time"
+
+import type { Device } from "lib/zod/device.ts"
+import type { NoiseThreshold } from "lib/zod/noise_threshold.ts"
+
+export const NOISE_SENSOR_DEVICE_TYPE = {
+  NOISEAWARE_ACTIVITY_ZONE: "noiseaware_activity_zone",
+  MINUT_SENSOR: "minut_sensor",
+} as const
+
+type NoiseSensorDeviceTypeFromMapping =
+  (typeof NOISE_SENSOR_DEVICE_TYPE)[keyof typeof NOISE_SENSOR_DEVICE_TYPE]
+
+export const NOISE_SENSOR_DEVICE_TYPE_LIST = Object.values(
+  NOISE_SENSOR_DEVICE_TYPE,
+) as NoiseSensorDeviceTypeFromMapping[]
+
+function formatAsNumber(
+  value: number | string | undefined,
+): number | undefined {
+  return Number.isNaN(Number(value)) ? undefined : Number(value)
+}
+
+function isCurrentTimeInRange(starts_daily_at: string, ends_daily_at: string) {
+  const starts_daily_at_zoned = ZonedTime.from(starts_daily_at)
+  const ends_daily_at_zoned = ZonedTime.from(ends_daily_at)
+
+  const now = DateTime.now().setZone(
+    starts_daily_at_zoned.getTimeZone().toString(),
+  )
+
+  const startTime = now.set({
+    hour: starts_daily_at_zoned.hour,
+    minute: starts_daily_at_zoned.minute,
+    second: starts_daily_at_zoned.second,
+  })
+
+  let endTime = now.set({
+    hour: ends_daily_at_zoned.hour,
+    minute: ends_daily_at_zoned.minute,
+    second: ends_daily_at_zoned.second,
+  })
+
+  if (endTime <= startTime) {
+    endTime = endTime.plus({ days: 1 })
+  }
+
+  if (startTime <= now && now <= endTime) {
+    return true
+  }
+
+  const nextDayStartTime = startTime.plus({ days: 1 })
+  if (now <= endTime && now >= nextDayStartTime) {
+    return true
+  }
+
+  return false
+}
+
+export const getCurrentlyTriggeringNoiseThresholds = ({
+  properties,
+  noise_thresholds,
+}: {
+  properties: Device["properties"]
+  noise_thresholds: NoiseThreshold[]
+}): string[] => {
+  let noise_level_decibels: number | undefined =
+    "noise_level_decibels" in properties
+      ? properties.noise_level_decibels
+      : undefined
+
+  const is_minut = "minut_metadata" in properties
+  const is_noise_aware = "noiseaware_metadata" in properties
+
+  if (noise_level_decibels == null) {
+    if (is_minut) {
+      noise_level_decibels = formatAsNumber(
+        properties?.minut_metadata?.latest_sensor_values?.sound?.value,
+      )
+    } else if (is_noise_aware) {
+      noise_level_decibels = formatAsNumber(
+        properties?.noiseaware_metadata?.noise_level_decibel,
+      )
+    } else {
+      noise_level_decibels = undefined
```

Comment:
Can this depend only on the value of `device.properties.noise_level_decibels` and not the provider?

Reply:
Good point—for the fake, that should be fine.